src/components/Login/Login.tsx

```diff
@@ -131,22 +130,19 @@ export default class LoginModal extends Component<Partial<LoginModalProps>, Logi
 
   public submitCredentials = async () => {
     const { form } = this.state;
-    const username = (form.username && form.username.value) || '';
-    const password = (form.password && form.password.value) || '';
+    const username = form.username?.value ?? '';
+    const password = form.password?.value ?? '';
     const { onSubmit } = this.props;
     if (onSubmit) {
       await onSubmit(username, password);
```

Comment:
While not an error, vscode suggested to remove the unnecessary `await`.
Turns out `onSubmit` was typed with `void` and not `Promise<void>` for its return value, fixed that as well
https://github.com/verdaccio/ui/blob/00d6223ad85f753c31e21e8eb3e85120d665f38a/src/components/Login/Login.tsx#L33

Reply:
well done.